### PR TITLE
Disarms have a higher chance against enemies with less health

### PR DESCRIPTION
**Pull request by Couls**
**What does this PR do:**
This PR makes it easier to disarm people who are more hurt and harder to disarm people who are less hurt. Disarms still have a 20% chance of happening against a full health target, and pushes still have a 10% chance of happening against a full health target. These numbers aren't set in stone i'd just prefer that disarming is less of an instant win button and requires some strategy or situational use. I'd prefer that people actually fight back instead of spamming disarm and hoping for a lucky roll. This doesn't take away RNG from disarms it just adds a slightly more reliable aspect and makes them even stronger in some cases

Disarms chances scale between 20-80%
and pushing over chances scale between 10-70% (this may go a bit too high, we'll have to see)
In comparison to our old system, our old system had a 25% chance to push someone over and a 60% chance to make them drop their weapon

**Changelog:**
:cl:
add: disarming now takes your opponents health into account
/:cl:

### CODE REVIEW

**Comment by Couls** [comment]
image

**Comment by Mitchs98** [comment]
Once again: Same as on Farie's PR. Disarm is fine as it is. It does not need to be changed. It most certainly does not need to be scales in favor of the attacker. Yes you can 'fight back' but the odds of you having a viable weapon to actually 'fight back' are far too low.

Disarming, and really combat in general in my opinion, as no reason whatsoever to be changed. Disarming especially as it's a decently important part to combat as a whole. It still does require 'strategy' and 'situational use' as it is now.

**Comment by Dinarzad** [comment]
Wow being an engineer sure is great. I like setting up solars.
Oh no...
Hark! What lies beyond this yonder shadow but a man, balding, clad in grey with a mask of gaseous protection upon his visage. Clad in his yellow mitted hands is a device of agony and pain, a device no honorable man ought grasp... a sword, crafted of energy!
This blackguard, this knave of ill-repute means to waylay me, and render my person asunder! Nay... NAY, SAYS I! I must needs get this vexatious blade from twixt his grip and flee, flee to the halls of the honorable constable, to scream aloud for aid!
Oh wait. I can't do that because I am *absolutely dead* because I couldn't disarm anything because I got jumped, shived, and the scaling disarm chance to knock whatever was out of my hand out of my grip and knock me over means every antag may as well have an innate stun baton grafted into the arms.

This is why these changes are getting backlash. Because they're short-sighted and are being made from a "But muh antag round..." perspective.  You are not owed a win as an antag. You are not owed the ability to just jump people an win the internet video game as an antag.
Sometimes things just go bad. It's SS13. Sometimes I also get killed by a vampire/changeling 5 minutes into the round and rendered perma-dead. OH well. It happens.  An antag losing his element of surprise cuz of disarm intent is right up there.
-1
There are no number changes that will make me change my mind here.  "Fixing" what was never broken has never gone well for any game ever, and I see no reason to think SS13 is any different.

**Comment by Shadeykins** [comment]
I think the primary issue people have with disarm is it makes anything remotely related to combat very nearly moot and a matter of "who can get the RNG roll." This coupled with bullets not having immediate stopping power in most cases with a short view range means that, even at range, disarms are a threat. The person can just run at you while you're shooting them and still come out on top.

**Comment by Dinarzad** [comment]
Which I understand but the thing is... I mean that's the game, more or less.  Look at something like DnD, you can have all the abilities of a god but it all, in the end, it frequently boils down to RNG. You can do amazingly OP spells/feats, but bad rolls can make you flub almost everything.

In more wholly SS13 senses however, the people in combat have tools to mitigate dice roll influences on what they are trying to achieve, it's why we have stun heavy combat systems. You can throw together stun prods which are 100% effective and knocking someone over, you can loot Stun batons from security, you can use flashes, or chemicals or any number of methods that are more effective then Disarm intent, because they're *not* based on a dice roll.   Disarm intent exists as that method of last resort that **everyone** has in their back pocket, it's something you know that the target or antag in question can do, so you need to or *should* plan for it.  If you fail to plan for it and that dice roll comes up, well... isn't that kind of your own fault?  If you try to solo yolo charge into a Xenomorph hive without a helmet should you really be surprised that it didn;t go well?  It was a risky play to not account for it and the risk sometimes doesn't pay off.  It sucks, I acknowledge that it does, to have an antag round swiftly ended because of it, but the alternative is much much worse, it's the least awful worst case that we have right now.

The PR's proposed not just here, but in the past, only account for *one* side of the spectrum, and that's the problem. Disarm intent is frequently the one and only way someone has even a snowball's chance at not being instantly dropped at first antag encounter.  Security has equipment to handle things but if you are jumped in maint, you're at an inherit disadvantage already, which is good, you *should* be.  But without disarm intent you may as well just nod politely at the changeling and buckle yourself into a chair for him, cuz you'll have absolutely nothing to defend yourself.  Changing disarm intent like has been attempted encourages more powergamey behaviour like keeping a stun prod in your bag at all times cuz you never know when you'll be jumped and this time, RNGesus is barely even an option anymore. A stunprod won't fail you, a 20% - 10% chance frequently will.

That's why I continue to oppose these changes, I understand why they're being made, but just slapping Disarm intent won't cut it, it seems like a small change, but it will lead to a chain reaction of other balance problems, it's not something that can just be done without a pre-established plan and roadmap, because like it or not, huge portions of the game are balanced around it existing in the state it does.

Additionally as a side-note on re-reading it: I apologize if my first response came across as sassy or in bad faith, it was mostly me just trying humorously reiterate what I've said before so it was less of a broken record, but in hindsight, it could come across as being a bit of a dick, which wasn't the intent.

**Comment by Couls** [comment]
I’m just trying to slowly reduce the amount of RNG in the system we have, disarms aren’t the biggest issue but it’s something that needs to be looked at alongside the one hit stuns because all these things serve to do is end combat swiftly and (arguably) unfairly like shadey said this helps move away from who gets the RNG roll. disarms are the first thing I want to modify because it’s something everyone has always and having the innate 60% chance of forcing anyone with a somewhat decent weapon to drop it doesn’t even the playing field it completely turns things over it’s the next best thing to a stunprod because it has a chance to stun and if that fails you still have the chance of disarming. It’s a two in one weapon and it makes using anything short of a guaranteed stun pointless

**Comment by Dinarzad** [comment]
>  I’m very open to feedback and I do understand where you’re coming from and why you think this wouldn’t be a good idea and I’d love for a compromise if we could brainstorm one where disarms aren’t strong weapons but great for self defense

And I'm open to that becoming a thing, make no mistake.  My problem is, what seems like a small change here is in actuality, not.  It's like trying to go in and fix atmospherics code.  That SOUNDS like it shouldn't be so bad on the surface, but in actuality that pile of spaghetti touches everything and pull one thread the wrong way and it all falls over.

In this instance that's not true on a code level but for balance.  I don't expect you to single handedly recode combat from the ground up, that's just not feasible, even for a small team of coders that's gonna take some serious effort.  But what I very much do want to see is a *plan*.  This plan need not be set in stone because, as you've said data might come up from playtesting, but an effort of this scale can't just be done on "Let's see what happens" because the potential to completely up-end the balance of antags, crew, security and various items is too present.  I've made similar arguments about why nerfing cloning or methods of revival need to be done very very carefully because it has wide-spread implications on balance.  Having a more ... lengthy discussion with several people and preferably staff members would be the ideal first step here, because our discussion right now could be moot depending on the direction they want Paradise's combat systems to go. 

 To do what you want to do, and I wholly understand why you want to do it, is going to take some pretty intensive overhauls. It'll probably be something that a group of people coordinate and work on, because I don't feel it can be done piecemeal without breaking a lot of things. That's mostly what I meant before, that this is the 'least awful' way we can work our presently stun-heavy combat system.

That said, since you asked what might be acceptable compromise from me, I am find with lowering the percent chance to push someone over to a degree.  I am also find with lowering the stun time for someone pushed over, how long they stay on the ground. Because to me, ideally, Disarm stance is to give yourself time to GTFO, or to do as the intent is called, disarm someone. Putting the disarm chance at a simple 50% and lowering the potential for a "Critical success" or lowering just how much a critical success it is, would be my suggestion. This, in my opinion, lowers the effectiveness of offensive Disarming, without reducing the effectiveness of defensively disarming all that much, as rushing to disarm an E-sword user is liable to get you stabbed in the pursuit of a low % stun, while knocking it out of his hands even for a click or two is enough time to start booking it.  You could knock it out of his hands to try an steal it, but that's some high risk-high reward balls to the walls plays goin on there (And a simple stun prod would be more effective as well.)

This is however, just my own personal opinion and thought process on the matter. Feel free to take it or ignore it as you see fit.

**Comment by Couls** [comment]
While lowering the percent chance might help the issue I feel like it’s a slippery slope between nerfing it to the point where disarms aren’t worth it anymore and everyone grabs stunprods or adjusting it so slightly no one will notice it’s been changed it’s kind of in a place where fine tuning it won’t make the issue better but I do agree the chances of success are stupidly high and that’s the whole point of this PR

**Comment by TDSSS** [comment]
> nerfing it to the point where disarms aren’t worth it anymore and everyone grabs stunprod

I mean...that doesn't sound too bad, imo. Weapons should give you an advantage to being barehanded. You shouldn't just rely on your ability to slap weapons out of people's hands, you're not some kind of kung fu ninja.

**Comment by Couls** [comment]
That's what this PR serves to do, it helps encourage using an actual weapon over disarming but still makes disarm viable in many situations

**Comment by Krysonism** [comment]
This PR does not solve disarm being a RNG crapshoot that makes melee weapons really unpleasant to use. I would suggest taking a look at this PR who solved the disarm problem in a beautiful way. https://github.com/tgstation/tgstation/pull/42958

It removes the chance element from disarm and instead makes it skill based, tactical and reliable.
It is possibly the the best PR of all time.

PS. As for stunprods, they are way too good on paradise, i suggest making them bulky so you can at least see if someone is carrying one on their back, or in their hand, and prepare.

**Comment by Couls** [comment]
> This PR does not solve disarm being a RNG crapshoot that makes melee weapons really unpleasant to use. I would suggest taking a look at this PR who solved the disarm problem in a beautiful way. [tgstation/tgstation#42958](https://github.com/tgstation/tgstation/pull/42958)
> 
> It removes the chance element from disarm and instead makes it skill based, tactical and reliable.
> It is possibly the the best PR of all time.
> 
> PS. As for stunprods, they are way too good on paradise, i suggest making them bulky so you can at least see if someone is carrying one on their back, or in their hand, and prepare.

Here's the issue with porting that over and the reason i've not been able to otherwise that is exactly what this PR would have been, this doesn't work without crawling. It just won't, being able to instantly stun someone against a wall or table would just make it stronger than the disarms are currently, this is kind of patching it over until I can make a PR for crawling which i'll start work on after diagonal movement is integrated, I would love for this to be our new system but its literally impossible with what we have in currently otherwise it would be a huge buff to disarm

**Comment by MrMagolor** [comment]
tl;dr of the walls of text above?

**Comment by Couls** [comment]
TGs disarm rework won’t work here because we have no crawling and you’ll be able to stunlock people against walls @mrmagolor

**Comment by MrMagolor** [comment]
>TGs disarm rework won’t work here because we have no crawling and you’ll be able to stunlock people against walls 

I never implied tg's rework at all?

**Comment by Couls** [comment]
No you asked for a tl;dr and I responded to someone who implied tg's rework
